utils/spawn.js

```diff
@@ -0,0 +1,21 @@
+var nodeSpawn = require('child_process').spawn;
+var Promise   = require('pinkie');
+var assign    = require('lodash').assign;
+
+
+module.exports = function (cmd, args, envs) {
+    return new Promise(function (resolve, reject) {
```

Comment:
Why not use on of many existing utility modules for this? E.g. https://github.com/inikulin/cp-sugar ?